virttest/test_setup.py

```diff
@@ -350,15 +343,15 @@ def check_hugepage_support(self):
         host_cpu_flags = utils_misc.get_cpu_flags()
         host_ker_cml = utils_misc.get_ker_cmd()
         if self.hugepage_cpu_flag not in host_cpu_flags:
-            raise HPNotSupportedError("Your host does not support hugepage, "
-                                      "as miss the cpu flag %s on your host."
-                                      "Please check cpu flags %s on the host" %
-                                      (self.hugepage_cpu_flag, host_cpu_flags))
+            self.test.cancel("Your host does not support hugepage, "
+                             "as miss the cpu flag %s on your host."
+                             "Please check cpu flags %s on the host" %
+                             (self.hugepage_cpu_flag, host_cpu_flags))
         if self.hugepage_match_str not in host_ker_cml:
-            raise HPNotSupportedError("Your host does not support hugepage, "
-                                      "as miss the %s in host kernel cmdline."
-                                      "Please check kernel cmdline %s on host" %
-                                      (self.hugepage_match_str, host_ker_cml))
+            self.test.cancel("Your host does not support hugepage, "
```

Comment:
can we use exceptions.TestSkipError() from framework so that we can avoid test here?